dogstatsd/src/datadog.rs

```diff
@@ -219,12 +234,65 @@ impl DdApi {
             }
         };
 
-        let resp = builder.send().await;
+        let resp = self.send_with_retry(builder, retry_strategy).await;
 
         let elapsed = start.elapsed();
         debug!("Request to {} took {}ms", url, elapsed.as_millis());
-        resp.map_err(|e| ShippingError::Destination(e.status(), format!("Cannot reach {url}")))
+        resp
     }
+
+    async fn send_with_retry(
+        &self,
+        builder: reqwest::RequestBuilder,
+        retry_strategy: RetryStrategy,
```

Comment:
`send_with_retry` also has access to `self`, do we really need to pass the `retry_strategy` here?

Reply:
Yup you're right, a holdover from when I initially passed it around